tests/syslog/test_syslog_source_ip.py

```diff
@@ -0,0 +1,560 @@
+        for k, v in mgmt_interface_info[mgmt_interface[0]].items():
+            ip_addr_instance = ip_address(unicode(k.split("/")[0]))
+            if isinstance(ip_addr_instance, IPv4Address):
+                MGMT_IP_ADDRESSES["ipv4"]["source_ip"] = k
+                for host in ip_network(unicode(v["gwaddr"])):
+                    if host != ip_addr_instance:
+                        syslog_server_ip = str(host)
+                        break
+                MGMT_IP_ADDRESSES["ipv4"]["syslog_server_ip"] = syslog_server_ip
+            elif isinstance(ip_addr_instance, IPv6Address):
+                MGMT_IP_ADDRESSES["ipv6"]["source_ip"] = k
+                for host in IPv6Network(unicode(v["gwaddr"])):
+                    if host != ip_addr_instance:
+                        syslog_server_ip = str(host)
+                        break
+                MGMT_IP_ADDRESSES["ipv6"]["syslog_server_ip"] = syslog_server_ip
+
+        logger.info("Updated mgmt dict:{}".format(MGMT_IP_ADDRESSES))
+        return mgmt_interface
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_ssip_test_env(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index,
+                            mgmt_interface, routed_interfaces, backup_and_restore_config_db_on_duts, localhost):
+        """
+        Setup env for ssip(syslog soruce ip) test
+        """
+        self.duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        self.asichost = self.duthost.asic_instance(enum_frontend_asic_index)
+
+        logger.info("Configure IP on the selected interface in default vrf")
+        self.configure_default_vrf_test_data(routed_interfaces)
+
+        logger.info("Enable mgmt vrf and configure IP on the mgmt interface")
+        self.configure_mgmt_vrf_test_data(localhost)
+
+        logger.info("Create data vrf and configure IP on the data vrf interface")
+        self.configure_data_vrf_test_data(routed_interfaces)
+
+    @pytest.fixture(scope="function", autouse=True)
+    def clear_syslog_config(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index):
+        """
+        clear syslog config for very test
+        """
+        self.duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+        yield
+        syslog_config_list = show_syslog(self.duthost)
+        if syslog_config_list:
+            for syslog_config in syslog_config_list:
+                del_syslog_server(self.duthost, syslog_server_ip=syslog_config["server ip"])
+
+    def configure_data_vrf_test_data(self, routed_interfaces):
+        """
+        Configure test data for data vrf
+        """
+        create_vrf(self.duthost, VRF_LIST[1])
+        bind_interface_to_vrf(self.asichost, VRF_LIST[1], routed_interfaces[1])
+        for k, v in DATA_VRF_IP_ADDRESSES.items():
+            self.asichost.config_ip_intf(routed_interfaces[1], DATA_VRF_IP_ADDRESSES[k]["source_ip"], "add")
```

Comment:
@JibinBao can we have all the commands being executed on a `DUT` + config dumps in `pytest` log with `INFO` severity?

Reply:
For all commands, there is log output in the pytest log, which is supported by the framework.

Reply:
@JibinBao The tests are usually being ran with INFO severity level. I didn't see these commands in the logs.
All the meaningful output should be present at INFO in front of DEBUG as well. This can simply test debugging a lot, especially when we are talking about configuration stage.

**P.S:** here i am mostly referring to statement like this:
```
logger.info("Update mgmt dict according to the real config on dut")
```

Reply:
Done